imports/ui/unit-explorer/filtered-units.jsx

```diff
@@ -0,0 +1,52 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import MenuItem from 'material-ui/MenuItem'
+import UnitTypeIcon from './unit-type-icon'
+
+import { resetMenuItemDivStyle } from '../general.mui-styles'
+
+export default class FilteredUnits extends Component {
+  render () {
+    const { unitsByRoles, searchResult, showAddBtn, handleUnitClicked, handleAddCaseClicked, administrate } = this.props
+    const filteredUnits = unitsByRoles || searchResult
```

Comment:
It seems to me that the only reason both `unitByRoles` and `searchResult` exists simultaneously is so the check whether the title for the unit list should appear. It's a very implicit way of doing it, which makes this component's API more ambiguous.
I suggest you merge both props into a single prop (name it `filteredUnits` maybe) and change `administrate` to some kind of an enum which holds 3 values for: 
* Showing 'Units I Administrate'
* Showing 'Units Involved In'
* Showing nothing